bundles/org.eclipse.swt/Eclipse SWT/common/org/eclipse/swt/internal/image/FileFormat.java

```diff
@@ -54,27 +54,47 @@ public abstract class FileFormat {
 		} catch (NoClassDefFoundError e) { } // ignore format
 	}
 
+	public static final int STATIC_ZOOM_FACTOR = 100;
+
+	static abstract class StaticFileFormat extends FileFormat {
+
+		abstract ImageData[] loadFromByteStream();
+
+		@Override
+		ImageData[] loadFromByteStream(int zoom) {
+			if (zoom == STATIC_ZOOM_FACTOR) {
+				return loadFromByteStream();
+			}
+			return Arrays.stream(loadFromByteStream()) //
+					.map(d -> {
+						int width = (int) Math.round(d.width * zoom / 100.);
+						int height = (int) Math.round(d.height * zoom / 100.);
+						return d.scaledTo(width, height);
```

Comment:
`ImageData#scaledTo()` has no shortcut returning itself when desired width/height are equal to original width/height, so this will always execute a scaling algorithm that in most cases will return a result equivalent to the input. We could also improve `ImageData#scaledTo()` to return itself in these cases, but that would also change the (implicit) contract of that method as you are not guaranteed to receive a new ImageData object anymore. So I would propose to check for zoom == 100 here and then simply return `d` instead of `d.scaledTo(...)`.

Reply:
> So I would propose to check for zoom == 100 here and then simply return d instead of d.scaledTo(...).

Isn't this already handled this way because the method calls and returns the result of `loadFromByteStream()` if `zoom == STATIC_ZOOM_FACTOR` at the top?

Reply:
Indeed, sorry, I missed that one. You have already perfectly covered that case 🙂